### ISSUE TITLE
Centos 7 buildbot frequently failing reflection and ccall tests

### ISSUE THREAD

**Issue by tkelman** [JuliaLang/julia#15878]
https://build.julialang.org/builders/build_centos7.1-x64?numbuilds=250

e.g. https://build.julialang.org/builders/build_centos7.1-x64/builds/693/steps/shell_2/logs/stdio (ccall failure doesn't happen every time, reflection failure has been)
```
	From worker 8:	     * ccall                Test Failed
	From worker 8:	  Expression: v < 10 ^ n
	From worker 8:	   Evaluated: 1000 < 1000
[snip]
	From worker 10:	     * reflection           Test Failed
	From worker 10:	  Expression: !(contains(str,"_$(i)"))
[snip]
Exception running test ccall :
On worker 8:
LoadError: There was an error during testing
 in record at ./test.jl:322
 in do_test at ./test.jl:220
 [inlined code] from ./test.jl:137
 in anonymous at ./<no file>:4294967295
 [inlined code] from ./essentials.jl:82
 in include_string at ./loading.jl:371
 in include_from_node1 at ./loading.jl:420
 [inlined code] from ./util.jl:179
 in runtests at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/testdefs.jl:7
 in #16 at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/runtests.jl:36
 in run_work_thunk at ./multi.jl:744
 [inlined code] from ./multi.jl:1038
 in #251 at ./event.jl:46
while loading /home/centos/buildbot/slave/build_centos7_1-x64/build/test/ccall.jl, in expression starting on line 506
Exception running test reflection :
On worker 10:
LoadError: There was an error during testing
 in record at ./test.jl:322
 in do_test at ./test.jl:220
 [inlined code] from ./test.jl:137
 in test_typed_ast_printing at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/reflection.jl:353
 [inlined code] from ./essentials.jl:82
 in include_string at ./loading.jl:371
 in include_from_node1 at ./loading.jl:420
 [inlined code] from ./util.jl:179
 in runtests at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/testdefs.jl:7
 in #16 at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/runtests.jl:36
 in run_work_thunk at ./multi.jl:744
 [inlined code] from ./multi.jl:1038
 in #251 at ./event.jl:46
while loading /home/centos/buildbot/slave/build_centos7_1-x64/build/test/reflection.jl, in expression starting on line 371
ERROR: LoadError: Some tests exited with errors.
 in (::##11#19)() at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/runtests.jl:64
 in cd(::##11#19, ::ASCIIString) at ./file.jl:48
 in include(::ASCIIString) at ./boot.jl:234
 in include_from_node1(::UTF8String) at ./loading.jl:417
 in process_options(::Base.JLOptions) at ./client.jl:263
 in _start() at ./client.jl:319
while loading /home/centos/buildbot/slave/build_centos7_1-x64/build/test/runtests.jl, in expression starting on line 13
```

**Comment by tkelman** [comment]
@yuyichao or @vtjnash do you have any ideas here? this is blocking nightlies

**Comment by yuyichao** [comment]
I added the `reflection` test. It is a little sensitive to the total number of local variables although I doubt this is the reason of the failure. I've started http://buildbot.e.ip.saba.us:8010/builders/build_centos7.1-x64/builds/744 to see what's wrong.

The `ccall` test looks like it's failing because all the works are done on the same worker thread? Maybe we should also compare it to the id of the current thread?
